src/utils/collections.zig

```diff
@@ -64,6 +64,188 @@ pub fn RecyclingList(
     };
 }
 
+pub fn SplitUnionList(TaggedUnion: type) type {
+    const Tag = @typeInfo(TaggedUnion).Union.tag_type.?;
+
+    return struct {
+        lists: sig.utils.types.EnumStruct(Tag, List),
```

Comment:
Would it be better to have an ArrayList(u8) thats bump allocated from? Where each item is `struct { Tag, DataType }` (user reads `Tag` first, then fieldParentPtr the struct to get the `Data` ptr).

Reply:
Can you think of a way to efficiently remove items? It's hard to imagine a good replacement for swapRemove with this approach.

Reply:
This is an interesting idea but it's not clear if this will work well. I made a ticket so we can look into it later instead of drawing out this pr: https://github.com/orgs/Syndica/projects/2/views/10?pane=issue&itemId=99205370